flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/common/TableIdRouter.java

```diff
@@ -95,4 +96,23 @@ private TableId resolveReplacement(
         }
         return TableId.parse(route.f1);
     }
+
+    public List<Set<TableId>> groupSourceTablesByRouteRule(Set<TableId> tableIdSet) {
```

Comment:
I doubt if it's really a generic and reusable method, without corresponding JavaDocs and test cases. Maybe just write it as a `for` loop in `SchemaDerivator`?

Reply:
Ditto.

Reply:
It makes use of "routes". In the latest code, I've added documentation for it.
Now that I think about it, it can also be placed in SchemaDerivator. I'll give it a try tonight.

Reply:
Sorry. The attempt couldn't be achieved. The current way of writing is more optimal.